src/bns-helpers.ts

```
@@ -0,0 +1,182 @@
+    const address = addressCV.address;
+
+    const price_function = properties.data['price-function'] as TupleCV;
+
+    const baseCV = price_function.data['base'] as IntCV;
+    const base: BN = baseCV.value;
+    const coeffCV = price_function.data['coeff'] as IntCV;
+    const coeff: BN = coeffCV.value;
+    const no_vowel_discountCV = price_function.data['no-vowel-discount'] as IntCV;
+    const no_vowel_discount: BN = no_vowel_discountCV.value;
+    const nonalpha_discountCV = price_function.data['nonalpha-discount'] as IntCV;
+    const nonalpha_discount: BN = nonalpha_discountCV.value;
+    const bucketsCV = price_function.data['buckets'] as ListCV;
+
+    const buckets: number[] = [];
+    const listCV = bucketsCV.list;
+    for (let i = 0; i < listCV.length; i++) {
+      const cv = listCV[i];
+      if (cv.type === ClarityType.UInt) {
+        buckets.push(cv.value);
+      }
+    }
+
+    const namespaceBNS: DbBNSNamespace = {
+      namespace_id: namespace,
+      address: addressToString(address),
+      base: base.toNumber(),
+      coeff: coeff.toNumber(),
+      launched_at: launched_at,
+      lifetime: lifetime.toNumber(),
+      no_vowel_discount: no_vowel_discount.toNumber(),
+      nonalpha_discount: nonalpha_discount.toNumber(),
+      ready_block: readyBlock,
+      reveal_block: revealed_at.toNumber(),
+      status: status,
+      latest: true,
+      buckets: buckets.toString(),
+      tx_id: txid,
+      canonical: true,
+      index_block_hash: indexBlockHash,
+    };
+    return namespaceBNS;
+  }
+
+  throw new Error('Invalid clarity type');
+}
+
+export async function parseContentHash(contentHash: string): Promise<string> {
+  let result: AttachmentValue | undefined = undefined;
+  try {
+    result = await new StacksCoreRpcClient().fetchJson<AttachmentValue>(
+      `v2/attachments/${contentHash}`,
+      {
+        method: 'GET',
+        timeout: 10 * 1000, //10 seconds
+      }
+    );
+  } catch (error) {}
```

Why hide the error here then throw below?

---

Updated this by throwing the error in the catch block.